src/fromKeys.ts

```diff
@@ -21,6 +21,14 @@ type FromKeys<T extends IterableContainer, V> = T extends readonly []
  * that key. Duplicate keys are overwritten, guaranteeing that `mapper` is run
  * for each item in `data`.
  *
+ * There are several other functions that could be used to build an object from
+ * an array:
+ * * `indexBy` - Builds an object from an array of *values* and a mapper for keys.
+ * * `pullObject` - Builds an object from an array of items with mappers for *both* keys and values.
+ * * `fromPairs` - Builds an object from an array of key-value pairs.
+ * * `mapToObj` - Builds an object from an array of items and a single mapper for key-value pairs.
+ * Refer to the docs for more details.
```

Comment:
**thought:** i wonder how we can make discoverability for these better...

* rxjs has the [operator decision tree](https://rxjs.dev/operator-decision-tree). "i have an array and i want to build an object"?
* ocaml plugin [merlin](https://github.com/ocaml/merlin/blob/master/doc/features.md) has a search feature by type; the syntax is `-list +string` for functions that have a list param and return a string
* haskell has [hoogle](https://hoogle.haskell.org/) and [rustdoc](https://github.com/rust-lang/rust/pull/23289) have other versions of searching by type, with other syntax

this ties into a wish for #519 , where we could somehow make mappings searchable

just thinking about this for now 🤷